pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -61,17 +210,203 @@ func (r *Builder) VirtualMachine(vmRef ref.Ref, object *cnv.VirtualMachineSpec,
+	if _, ok := vm.Image.Properties[PointerModel]; ok {
+		tablet := cnv.Input{
+			Type: Tablet,
+			Name: Tablet,
+			Bus:  USB,
+		}
+		object.Template.Spec.Domain.Devices.Inputs = []cnv.Input{tablet}
+	}
+}
+
+func (r *Builder) mapCPUMem(vm *model.Workload, object *cnv.VirtualMachineSpec) {
+	// Set MachineType
+	machineType := DefaultProperties[MachineType]
+	if imageMachineType, ok := vm.Image.Properties[MachineType]; ok {
+		machineType = imageMachineType.(string)
+	} else {
+		for _, volume := range vm.Volumes {
+			if volume.Bootable == "true" {
+				if volumeMachineType, ok := volume.VolumeImageMetadata[MachineType]; ok {
+					machineType = volumeMachineType
+				}
+			}
+		}
+	}
+	object.Template.Spec.Domain.Machine = &cnv.Machine{Type: machineType}
+
+	// Set CPU Policy
+	cpuPolicy := DefaultProperties[CPUPolicy]
+	if imageCPUPolicy, ok := vm.Image.Properties[CPUPolicy]; ok {
+		cpuPolicy = imageCPUPolicy.(string)
+	}
+	if cpuPolicy == CPUPolicyDedicated {
+		object.Template.Spec.Domain.CPU.DedicatedCPUPlacement = true
+	}
+
+	// Set CPU Thread Policy
+	cpuThreadPolicy := DefaultProperties[CPUThreadPolicy]
+	if imageCPUThreadPolicy, ok := vm.Image.Properties[CPUThreadPolicy]; ok {
+		cpuThreadPolicy = imageCPUThreadPolicy.(string)
+	}
+	if cpuThreadPolicy == CPUThreadPolicyIsolate {
+		object.Template.Spec.Domain.CPU.IsolateEmulatorThread = true
+	}
+
+	// Set CPU Sockets/Cores/Threads and Memory requests
+	var cpuSockets, cpuCores, cpuThreads int
+
+	cpuSockets = vm.Flavor.VCPUs
+	cpuCores = 1
+	cpuThreads = 1
+
+	if flavorCPUSockets, ok := vm.Flavor.ExtraSpecs[FlavorCPUSockets]; ok {
+		if sockets, err := strconv.Atoi(flavorCPUSockets); err == nil {
+			cpuSockets = sockets
+		}
+	} else if imageCPUSockets, ok := vm.Image.Properties[CPUSockets]; ok {
+		if sockets, ok := imageCPUSockets.(int); ok {
+			cpuSockets = sockets
+		}
+	}
+	if flavorCPUCores, ok := vm.Flavor.ExtraSpecs[FlavorCPUCores]; ok {
+		if cores, err := strconv.Atoi(flavorCPUCores); err == nil {
+			cpuCores = cores
+		}
+	} else if imageCPUCores, ok := vm.Image.Properties[CPUCores]; ok {
+		if cores, ok := imageCPUCores.(int); ok {
+			cpuCores = cores
+		}
+	}
+	if flavorCPUThreads, ok := vm.Flavor.ExtraSpecs[FlavorCPUThreads]; ok {
+		if threads, err := strconv.Atoi(flavorCPUThreads); err == nil {
+			cpuThreads = threads
+		}
+	} else if imageCPUThreads, ok := vm.Image.Properties[CPUThreads]; ok {
+		if threads, ok := imageCPUThreads.(int); ok {
+			cpuThreads = threads
+		}
+	}
+
+	resourceRequests := map[core.ResourceName]resource.Quantity{}
+
+	if cpuCores != 1 || cpuThreads != 1 {
+		object.Template.Spec.Domain.CPU.Sockets = uint32(cpuSockets)
+		object.Template.Spec.Domain.CPU.Cores = uint32(cpuCores)
+		object.Template.Spec.Domain.CPU.Threads = uint32(cpuThreads)
+	} else {
+		cpuRequest := resource.NewQuantity(int64(cpuSockets*cpuCores*cpuThreads), resource.BinarySI)
+		resourceRequests[core.ResourceCPU] = *cpuRequest
+	}
```

Comment:
don't we need to set both the cpu topology in `object.Template.Spec.Domain.CPU` and `core.ResourceCPU` on the pod?

Reply:
According to this [documentation](https://kubevirt.io/user-guide/virtual_machines/dedicated_cpu_resources/) we should use either `Spec.Domain.CPU` or `Spec.Domain.Resources.[Requests,Limits], but not both at the same time.

Reply:
ack, but then it's better to use Spec.Domain.CPU as it should affect the cpu topology of the guest